### PR TITLE
(closes #1336) add GHA to check MD links

### LINKED ISSUE #1336: Have GHA check for outdated links in all markdown files.
@LonelyCat124  has reported:

The contributing.md in PSyclone links to "Working with PSyclone from GitHub", but that link appears to be broken (https://psyclone.readthedocs.io/en/stable/developers.html#working-with-psyclone-from-github) - i think its moved to https://psyclone-dev.readthedocs.io/en/latest/working_practises.html but I'm not sure if there are other broken links too

### PR DESCRIPTION
**Pull request by arporter**
Adds a new action to GitHub Actions that checks all links in all markdown files.
Fixes various broken links in said markdown files.
Brings parts of the dev guide up-to-date and documents the new GitHub Action (under Continuous Integration).

### CODE REVIEW

**Comment by arporter** [comment]
The link checker correctly reports the dead link in CONTRIBUTING.md. However, it also finds lots of other dead links which I will proceed to fix...

**Comment by arporter** [comment]
This is ready for first review, subject to CI happiness. Could be one for @TeranIvy or @hiker?

File: doc/developer_guide/working_practises.rst [hunk 1]
```diff
@@ -69,12 +69,11 @@ repository then doing::
 
   > cd <PSYCLONEHOME>
   > git submodule init
-  > git submodule update --init --recursive
+  > git submodule update --init
```
Comment (hiker) [inline doc/developer_guide/working_practises.rst:72]: Good catch that!

File: doc/developer_guide/working_practises.rst [hunk 2]
```diff
@@ -348,11 +352,11 @@ and therefore the line described above must be commented out again
...
 A single run of the test suite on GitHub Actions uses
-approximately 15 minutes of CPU time and we run the test suite on three
-different versions of Python. Therefore, ideally we would avoid
+approximately 20 minutes of CPU time and we run the test suite on three
+different versions of Python. Therefore, it is good practise to avoid
 triggering the tests unnecessarily (e.g. when we know that a certain commit
-won't pass). Unfortunately we have yet to find a working solution
-for this - see Issue #973.
+won't pass). This may be achieved by including the "[skip ci]" tag (without
+the quotes) in the associated commit message.
```
Comment (hiker) [inline doc/developer_guide/working_practises.rst:359]: Great!

File: tutorial/practicals/LFRic/building_code/3_time_evolution/README.md [hunk 3]
```diff
@@ -67,11 +67,11 @@ no modifications are:
   outputs results calling an I/O routine from the
   `write_diagnostics_mod.f90` module;
 
-* [`write_diagnostics_mod.f90`](write_diagnostics_mod.f90) calls a diagnostic
-  output routine to write the "model state" consisting of the coordinate and
-  perturbation fields to a file;
+* [`write_diagnostics_mod.f90`](../gungho_lib/write_diagnostics_mod.f90)
```
Comment (hiker) [inline tutorial/practicals/LFRic/building_code/3_time_evolution/README.md:70]: Really great to get them all fixed!

**Review by hiker — COMMENTED** [review]
All good (how can it not be ... the first commit to ever get 6 green ticks :) ), I've brought it up to current master and will merge it now.

**Review by hiker — APPROVED** [review]
Ready to go in :)

**Pull request merged**